docs/sort-filepairs-by-samplename.md

```diff
@@ -0,0 +1,25 @@
+## Problem 
+
+You have many tuples with a sample name that works as an identifier and you want to sort by the sample name.
+
+## Solution
+
+Use the [toSortedList](https://www.nextflow.io/docs/latest/operator.html#tosortedlist) and the [flatMap](https://www.nextflow.io/docs/latest/operator.html#flatmap) operators to convert the channel to a sorted list, and then convert back to the original structure you get from the [fromFilePairs](https://www.nextflow.io/docs/latest/channel.html?highlight=fromfilepairs#fromfilepairs) channel factory.
+
+## Code
+
+```groovy
+Channel
+  .fromFilePairs('/my/path/data/*_{1,2}.fastq')
+  .toSortedList( { a, b -> a[0] <=> b[0] } )
+  .flatMap()
+  .view()
```

Comment:
I think this is broken, `a[0]` should take this first segment in the file path i.e. `some` in `/some/data/file.name`. 

I think you want to do `a.name <=> b.name`

Also `.toSortedList( { a, b -> a.name <=> b.name } )` can be simplified either with 


```
.toSortedList{ a, b -> a.name <=> b.name }
```

Reply:
Not also that `.toSortedList{ a, b -> a <=> b }` would be possible and it would sort them against the full file path, that in some cases makes sense

Reply:
Broken in what sense? Not working? It worked in my machine 😬 See [here](https://nextflow.slack.com/archives/C02T98A23U7/p1670341624277339?thread_ts=1670337268.847439&cid=C02T98A23U7).

Reply:
So... What do you think, @pditommaso ? 😬

Reply:
You are right. I think however the snippet can be more useful by adding a comment for each line explaining what it does otherwise it's not clear. 

Also, worth to mentioning that `toSortedList` is not scalable because it introduces a blocking point in the pipeline execution since to sort the entries it needs to collect all of them before

Reply:
Very good points! Thanks, I will work on it 😄